apps/rxjs/race-condition/src/app/app.component.ts

```diff
@@ -1,37 +1,42 @@
 import {
   ChangeDetectionStrategy,
   Component,
+  computed,
   inject,
-  OnInit,
 } from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
 import { MatDialog } from '@angular/material/dialog';
 import { take } from 'rxjs';
 import { TopicModalComponent } from './topic-dialog.component';
-import { TopicService, TopicType } from './topic.service';
+import { TopicService } from './topic.service';
+import { NgIf } from '@angular/common';
 
 @Component({
   standalone: true,
   selector: 'app-root',
-  template: ` <button (click)="openTopicModal()">Open Topic</button> `,
+  imports: [NgIf],
+  template: `
+    <ng-container *ngIf="isLoading$(); else loading">
+      <button (click)="openTopicModal()">Open Topic</button> .
+    </ng-container>
+    <ng-template #loading> loading... </ng-template>
+  `,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AppComponent implements OnInit {
+export class AppComponent {
   title = 'rxjs-race-condition';
   dialog = inject(MatDialog);
   topicService = inject(TopicService);
-  topics: TopicType[] = [];
+  topics = toSignal(this.topicService.fakeGetHttpTopic().pipe(take(1)), {
+    initialValue: [],
+  });
 
-  ngOnInit(): void {
-    this.topicService
-      .fakeGetHttpTopic()
-      .pipe(take(1))
-      .subscribe((topics) => (this.topics = topics));
-  }
+  isLoading$ = computed(() => this.topics().length > 0);
```

Comment:
That's not a good condition for the loading indicator. What is topic length is zero ?